app/src/main/java/ru/myuniquenickname/myapplication/presentation/movieList/MovieListFragment.kt

```diff
@@ -118,12 +125,61 @@ class MovieListFragment : Fragment() {
     }
 
     private fun search() {
-        binding.searchButton.setOnClickListener {
-            val textSearch = binding.editMovieText.text.toString()
-            if (textSearch.isNotEmpty()) {
-                movieListViewModel.searchMovie(textSearch)
+
+        binding.editMovieText.doAfterTextChanged {
+            lifecycleScope.launch {
+                movieListViewModel.queryChannel.send(it.toString())
             }
         }
+        movieListViewModel.moviesListSearch.observe(
+            this,
+            {
+                when (it) {
+                    is ValidResult -> {
+                        adapterMovies.submitList(it.result)
+                        binding.spinner.visibility = View.INVISIBLE
+                    }
+                    is ErrorResult -> {
```

Comment:
Boilerplate code. You should create a function which receives string resource and show a toast.